engineering/conventions-go.md

```diff
@@ -69,3 +64,9 @@ A blank line is usually added after the end of any control structure, too.
 ### Naming
 
 * Prefer full names (e.g. `Repository`, not `Repo`).
+
+## CLI
+
+* Put sources for your executable commands in `cmd/<command-name>/`, with the `main` function in `cmd/<command-name>/main.go`.
+* We use [go-flags](https://github.com/jessevdk/go-flags) extensively for CLI options parsing.
```

Comment:
I still think we should use something less magical. If some flag is not being set properly from CLI it's pretty hard to see why because everything is hidden in reflection code.

I don't have any alternative to propose right now (except `cobra`, which is far less convenient), so I will leave my concerns for now. Will raise them a bit later when I have time to show an alternative that is both convenient to use and easy to debug.

Reply:
https://github.com/ogier/pflag ?